packages/data/src/plugins/actions.ts

```diff
@@ -348,6 +349,36 @@ export function setRecommendedPlugins(
 	};
 }
 
+const SUPPORTED_TYPES = [ 'payments' ];
+export function* dismissRecommendedPlugins( type: RecommendedTypes ) {
+	if ( ! SUPPORTED_TYPES.includes( type ) ) {
+		return [];
+	}
+	const plugins: Plugin[] = yield resolveSelect(
+		STORE_NAME,
+		'getRecommendedPlugins',
+		type
+	);
+	yield setRecommendedPlugins( type, [] );
```

Comment:
Optimistically dismiss the recommend plugins, if the below request fails we set the recommended plugins back